lib/src/table/table.service.ts

```diff
@@ -0,0 +1,436 @@
+        }
+
+        if (paginatedTable.length === 0) {
+            paginatedTable.push([]);
+        }
+
+        return paginatedTable;
+    }
+
+    /**
+     * SETUP TABLE HEADER:
+     * Sets up the table header according to the current master table and it's config info
+     * @param {object} types a REQUIRED map of every column name what type of data it represents
+     * @param {object} labels an OPTIONAL map of column names and what label to display as column
+     */
+    private setupTableHeader(types: TableConfig<T>["types"], labels?: TableConfig<T>["labels"]): void {
+        const tableHeaderList: TableHeaderListItem<T>[] = [];
+
+        for (const columnName of this.columnsList) {
+            const tableKeySelector: keyof T = columnName as keyof T;
+            const label: string = labels && labels[columnName] ? labels[columnName] : readableFromCamelCase(columnName);
+            const valueType: TableHeaderListValueType = types[columnName] || "string";
+            tableHeaderList.push({
+                tableKeySelector,
+                valueType,
+                label,
+            });
+        }
+        this._tableHeaderList = tableHeaderList;
+    }
+
+    /**
+     * Sort the table:
+     *
+     * This method will sort the table based on the column selected and it's ascending/descending property
+     * @param {TableRowWithMeta<T>[]} table the table to be sorted with meta data
+     * @param {SortInfo<keyof T>} sortInfo The information on how to sort the table: column name, type and asc/desc
+     * @returns the sorted table
+     */
+    private sortTable = (table: TableRowWithMeta<T>[], sortInfo: SortInfo): TableRowWithMeta<T>[] => {
+        const { column, isAscending, type }: SortInfo<keyof T> = sortInfo;
+        const newSortedTable: TableRowWithMeta<T>[] = table.sort((a: TableRowWithMeta<T>, b: TableRowWithMeta<T>) => {
+            // flipping the first and second items based on ascending/descending
+            const firstItem: any = isAscending ? a.row[column] : b.row[column];
+            const secondItem: any = isAscending ? b.row[column] : a.row[column];
+            switch (type) {
+                case "string":
+                    return String(firstItem).toLowerCase().localeCompare(String(secondItem).toLowerCase());
+                case "date":
+                    return toDate(firstItem) > toDate(secondItem) ? 1 : -1;
+                default:
+                    return (firstItem as any) - (secondItem as any);
+            }
+        });
+
+        return [...(newSortedTable ? newSortedTable : [])];
+    };
+
+    /**
+     * CHECK IS ALL SELECTED
+     * Checks the selected rows of each table and check returns true/false if all rows in every table are selected
+     * @returns {boolean}
+     */
+    private checkIsAllSelected(): boolean {
+        return this._selectedRows?.map((e, i) => e?.length === this._paginatedTable[i]?.length).every(e => e === true);
+    }
```

Comment:
If `e` is boolean, you don't need to compare it to `true`
```suggestion
        return this._selectedRows?.map((e, i) => e?.length === this._paginatedTable[i]?.length).every(e => e);
    }
```